datafusion/catalog-listing/src/helpers.rs

```diff
@@ -121,40 +120,6 @@ pub fn expr_applicable_for_cols(col_names: &[&str], expr: &Expr) -> bool {
 /// The maximum number of concurrent listing requests
 const CONCURRENCY_LIMIT: usize = 100;
 
-/// Partition the list of files into `n` groups
-pub fn split_files(
```

Comment:
this is an API change -- can you perhaps leave the function here and mark it `#deprecated` per https://datafusion.apache.org/contributor-guide/api-health.html#deprecation-guidelines

Reply:
done https://github.com/apache/datafusion/pull/15379/commits/7415f7c6846b33136ee0c833c6836a3c97508b6b